modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -723,10 +723,11 @@ void ONNXImporter::populateNet()
 static
 const std::string& extractNodeName(const opencv_onnx::NodeProto& node_proto)
 {
-    if (node_proto.has_name() && !node_proto.name().empty())
-    {
-        return node_proto.name();
-    }
+//    We need to rework DNN outputs API first, this causes #21698 right now
+//    if (node_proto.has_name() && !node_proto.name().empty())
+//    {
+//        return node_proto.name();
+//    }
     for (int i = 0; i < node_proto.output_size(); ++i)
```

Comment:
Problem is still here.
There are several "outputs" with different names. Why do we pick the first non-empty output name?

IMHO, we should register all "outputs" instead.

Reply:
@rogday Could you please prepare a small test case for this problem? I will try to fix it

Reply:
@alalek, sure, you could use tests at https://github.com/opencv/opencv_extra/pull/968.